### PR TITLE
New ParametersTest file

### PR DESCRIPTION
**Pull request by piRGoif**
Adding a more atomic test of the PR Combodo/itop-data-collector-base#9
See https://github.com/Combodo/itop-data-collector-base/pull/9#issuecomment-1590803158

### CODE REVIEW

File: test/ParametersTest.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * @copyright   Copyright (C) 2010-2023 Combodo SARL
+ * @license     http://opensource.org/licenses/AGPL-3.0
+ */
+
+namespace UnitTestFiles\Test;
+
+use Parameters;
+use PHPUnit\Framework\TestCase;
+
+@define('APPROOT', dirname(__FILE__, 2).'/');
```
Comment (Hipska) [inline test/ParametersTest.php]: What about
```suggestion
@define('APPROOT', dirname(__DIR__).'/');
```

**Review by Hipska — CHANGES_REQUESTED** [review]

**Review by Hipska — COMMENTED** [review]
https://github.com/Combodo/itop-data-collector-base/pull/9#issuecomment-1543679526
> The only thing I'm having a problem with is that your test requires the xml to be exactly the same. While an xml with elements in a different order would result in the same configuration as expected, but the test would fail. Both xml and php arrays don't have a guaranteed fixed ordering of elements.
> 
> I would be fine if the test would say for example it should contain `<broken_param>(&amp;(objectClass=person)(mail=*))</broken_param>`.

**Comment by piRGoif** [comment]
I'm not sure I understand your comment, as the tests uses assertStringContainsString() ?

**Comment by Hipska** [comment]
This test does, but the one I was referring to in my original comment (test/UtilsTest.php) isn't.

Also, the second test here for the arrays might sometimes fail, but I guess it is fine for now.

**Comment by piRGoif** [comment]
Ok, thanks for the merge !

**Pull request merged**